### ISSUE TITLE
Remote: Iteration Plan for August 2022

### ISSUE THREAD

**Issue by kieferrm** [microsoft/vscode-remote-release#7033]
This plan captures our work in **August**. We follow the same iteration cycle as [VS Code](https://github.com/microsoft/vscode/issues/157454). Although we plan for a whole iteration, we may not only ship at the end of an iteration but throughout.

<details>
<summary>Legend of annotations</summary>

| Mark                 | Description                                       |
| -------------------- | ------------------------------------------------- |
| 🏃‍♀️                   | work in progress                                  |
| ✋                   | blocked task                                      |
| 💪                   | stretch goal for this iteration                   |
| 🔴                   | missing issue reference                           |
| 🔵                   | more investigation required to remove uncertainty |
| ⚫                   | under discussion within the team                  |
| ⬛                   | a large work item, larger than one iteration      |

</details>

## Getting started

- [x] Come up with a plan about how to decide what a Getting Started flow ala WWBD should be like; [issue](https://github.com/microsoft/vscode-python/issues/19676); @brettcannon

## Environments

- [ ] 🏃‍♀️ Work on finalizing design of proposed API for environments; [issue](https://github.com/microsoft/vscode-python/issues/19101); @karrtikr

## Editing

None

## LSP

- [x] Convert `lsprotocol` to snake case; [issue](https://github.com/microsoft/vscode-python/issues/19652); @karthiknadig

## Terminal

None

## Debugging

- [ ] 🏃‍ Create a plan to break out debugging support into it's own extension; [issue](https://github.com/microsoft/vscode-python/issues/19503) @paulacamargo25 @karthiknadig

## Formatting

- [ ] 🏃‍♀️ Publish an `autopep8` extension; [issue](https://github.com/microsoft/vscode-python/issues/19654); @karthiknadig

## Linting

- [x] Publish a flake8 extension; [issue](https://github.com/microsoft/vscode-python/issues/19655); @karthiknadig

## Testing

- [x] Rewrite `unittest` execution code; [issue](https://github.com/microsoft/vscode-python/issues/19098); @ksy7588 @karthiknadig
- [ ] 🏃‍♀️ Investigate how to do pytest execution/discovery via plug-in; [issue](https://github.com/microsoft/vscode-python/issues/19432); @ksy7588 @karthiknadig

## `(github|vscode).dev`

None

## Documentation

None

## Engineering

- [ ] ✋ Write web tests; [issue](https://github.com/microsoft/vscode-python/issues/19115); @karthiknadig

## Python community

- [x] Write down a plan for what needs to happen for healthy support for WASI; [issue](https://github.com/microsoft/vscode-python-internalbacklog/issues/438); @brettcannon

---

<details>
<summary>Deferred Items</summary>

- [ ] Explore options for activated terminals and terminal profiles [vscode-python#11039](https://github.com/microsoft/vscode-python/issues/11039) @luabud @karrtikr
- [ ] Control rollout of notifying about the Black extension; [issue](https://github.com/microsoft/vscode-python/issues/19653) ; @luabud @karthiknadig
- [ ] Control rollout of notifying about the isort extension; [issue](https://github.com/microsoft/vscode-python/issues/19653); @luabud @karthiknadig
- [ ] Control rollout of notifying about the Pylint extension; [issue](https://github.com/microsoft/vscode-python/issues/19653); @luabud @karthiknadig
- [ ] ✋ 🔵 Work w/ Pylance on automatic indentation [vscode-python#481](https://github.com/microsoft/vscode-python/issues/481) @brettcannon (blocked on [VS Code issue](https://github.com/microsoft/vscode/issues/131707))
- [ ] ✋ Create platform-specific VSIXs (blocked on standardizing on extension pipelines supporting platform-specific VSIXs); [vscode-python#17439](https://github.com/microsoft/vscode-python/issues/17439) @paulacamargo25
- [ ] Infer environment variables for an activated conda environment [vscode-python#17332](https://github.com/microsoft/vscode-python/issues/17332) @karrtikr
- [ ] Rewrite test discovery for pytest [vscode-python#17242](https://github.com/microsoft/vscode-python/issues/17242) @kimadeline
- [ ] Rewrite test execution for pytest
- [ ] ✋ Start work on implementing `packaging.metadata`; [issue](https://github.com/pypa/packaging/issues/383); @brettcannon
- [ ] ✋ Finalize Python Launcher plug-in design (blocked on [conda's feedback](https://github.com/conda/conda/issues/11283)); [discussion](https://github.com/brettcannon/python-launcher/discussions/168); @brettcannon
- [ ] Support `.python-version` for pyenv; [issue](https://github.com/microsoft/vscode-python/issues/1167); @karrtikr
- [ ] Investigate any remaining places we are not using `conda run`
- [ ] Implement a Python traceback problem matcher in core
- [ ] 🔴 Use `conda run` when debugging (waiting on the conda team to give us the go-ahead to rely on `conda run` so fundamentally); @paulacamargo25
- [ ] Provide settings to control what environment details are displayed; @karrtikr
- [ ] Investigate configuring running and debugging

</details>
